charts/backbeat/templates/lifecycle/producer_deployment.yaml

```diff
@@ -0,0 +1,74 @@
+apiVersion: extensions/v1beta1
```

Comment:
This looks like 3x almost-the-same file. Consider abstracting out the template, parametrize it and run it 3 times?

Reply:
Definitely agree with Nicolas but lets leave that for another PR. I wanted to separate the burden of the refactor and the addition of this feature. We have a ticket ZENKO-388 specifically to do exactly what Nicolas is speaking of. 😄